variables.tf

```diff
@@ -77,82 +78,151 @@ variable "cluster_name" {
+#-------------------------------
+# EKS Cluster Security Groups
+#-------------------------------
+variable "cluster_additional_security_group_ids" {
+  description = "List of additional, externally created security group IDs to attach to the cluster control plane"
+  type        = list(string)
+  default     = []
 }
 
-variable "cluster_endpoint_private_access" {
+variable "cluster_security_group_additional_rules" {
+  description = "List of additional security group rules to add to the cluster security group created. Set `source_node_security_group = true` inside rules to set the `node_security_group` as source"
+  type        = any
+  default     = {}
+}
+#-------------------------------
+# EKS Cluster VPC Config
+#-------------------------------
+variable "cluster_endpoint_public_access" {
   type        = bool
-  default     = false
-  description = "Indicates whether or not the EKS private API server endpoint is enabled. Default to EKS resource and it is false"
+  default     = true
+  description = "Indicates whether or not the EKS public API server endpoint is enabled. Default to EKS resource and it is true"
 }
 
-variable "cluster_create_endpoint_private_access_sg_rule" {
+variable "cluster_endpoint_private_access" {
   type        = bool
   default     = false
-  description = "Whether to create security group rules for the access to the Amazon EKS private API server endpoint"
+  description = "Indicates whether or not the EKS private API server endpoint is enabled. Default to EKS resource and it is false"
 }
 
-variable "cluster_endpoint_private_access_cidrs" {
+variable "cluster_endpoint_public_access_cidrs" {
+  description = "List of CIDR blocks which can access the Amazon EKS public API server endpoint"
   type        = list(string)
-  default     = []
-  description = "List of CIDR blocks which can access the Amazon EKS private API server endpoint"
+  default     = ["0.0.0.0/0"]
 }
 
-variable "cluster_endpoint_private_access_sg" {
-  type        = list(string)
-  default     = []
-  description = "List of security group IDs which can access the Amazon EKS private API server endpoint"
+#-------------------------------
+# EKS Cluster ENCRYPTION
+#-------------------------------
+variable "cluster_kms_key_arn" {
+  type        = string
+  default     = null
+  description = "A valid EKS Cluster KMS Key ARN to encrypt Kubernetes secrets"
 }
 
-variable "cluster_endpoint_public_access" {
-  type        = bool
-  default     = true
-  description = "Indicates whether or not the EKS public API server endpoint is enabled. Default to EKS resource and it is true"
+variable "cluster_kms_key_deletion_window_in_days" {
+  type        = number
+  default     = 30
+  description = "The waiting period, specified in number of days (7 - 30). After the waiting period ends, AWS KMS deletes the KMS key"
 }
 
-variable "cluster_endpoint_public_access_cidrs" {
+variable "cluster_encryption_config" {
```

Comment:
variable is required but not being used in the project, probably wanted to use it in `main.tf` ?

Reply:
updated with condition